github-upload-action-secrets

```diff
@@ -46,10 +51,24 @@ def encrypt(public_key: str, secret_value: str) -> str:
 
     public_key = public.PublicKey(public_key.encode("utf-8"), encoding.Base64Encoder())
     sealed_box = public.SealedBox(public_key)
-    encrypted = sealed_box.encrypt(secret_value.encode("utf-8"))
+    encrypted = sealed_box.encrypt(secret_value)
     return b64encode(encrypted).decode("utf-8")
 
 
+def generate_ssh_key(pubkey):
+    """Generates a Ed25519 keypair and outputs public/private in OpenSSH format"""
+    key = Ed25519PrivateKey.generate()
+    public = key.public_key().public_bytes(Encoding.OpenSSH, PublicFormat.OpenSSH)
+    private = key.private_bytes(Encoding.PEM, PrivateFormat.OpenSSH, NoEncryption())
+
+    return public, encrypt(pubkey["key"], private)
+
+
+def read_encrypted(filename, pubkey):
```

Comment:
I find this naming a bit confusing -- it does *not* read an encrypted file, it reads a plaintext file and encrypts it. `read_and_encrypt()` or just `read_encrypt()`?

Reply:
Sure.